### PR TITLE
One new C function, several documentation changes

### PR DESCRIPTION
**Pull request by plredmond**
Hi! This is a random grab-bag of documentation improvements. In addition to that, I've exposed exactly one additional C API function.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      sha256 = "1cl4ka4kk7kh3bl78g06dhiidazf65q8miyzaxi9930d6gwyzkci";
+    })
+    { inherit config; };
+  # package set for haskell compiler version
+  haskellCompilerPackages = nixpkgs.haskellPackages; # don't care about compiler version
+  # override package set to inject project components
+  haskellPackages = haskellCompilerPackages.override
+    (old: {
+      overrides = self: super: with nixpkgs.haskell.lib; rec {
+        gomp = self.hello;
```
Comment (plredmond) [inline default.nix:19]: This is definitely not correct for OSX or MacOS users. It might be better to leave out `default.nix` and `makefile` from the PR.

**Review by IagoAbal — COMMENTED** [review]
Yes please leave `.gitignore`, `default.nix`, and `makefile` out of the PR. But I am happy to take the other changes. Thanks!

File: default.nix (hunk 1 above)
Comment (plredmond) [reply default.nix]: They are deleted.

**Comment by plredmond** [comment]
I've reverted the commit which added those files, and also cherry-picked another commit which adds a few more functions. I don't know if I added them in the "right" place vertically in the file, but otherwise followed the conventions.

**Review by IagoAbal — APPROVED** [review]
Thanks!

**Pull request merged**